ledger/src/blockstore.rs

```diff
@@ -3847,8 +3873,8 @@ fn slot_has_updates(slot_meta: &SlotMeta, slot_meta_backup: &Option<SlotMeta>) -
     // We should signal that there are updates if we extended the chain of consecutive blocks starting
     // from block 0, which is true iff:
     // 1) The block with index prev_block_index is itself part of the trunk of consecutive blocks
-    // starting from block 0,
-    slot_meta.is_connected() &&
+    // starting from block 0 or from a snapshot, which is rooted by definition
+    slot_meta.is_parent_connected() &&
         // AND either:
         // 1) The slot didn't exist in the database before, and now we have a consecutive
         // block for that slot
```

Comment:
A bit unrelated to this PR.  I feel the comment could be further simplified as the real condition below is simply whether the consumed count is updated (i.e., `slot_meta_backup.as_ref().unwrap().consumed != slot_meta.consumed`).

However, we need to break into two conditions and further check `slot_meta_backup.is_none()` because if slot_meta_backup doesn't exist, then we cannot check `slot_meta_backup.as_ref().unwrap().consumed != slot_meta.consumed` because it does not exist.

Reply:
I ended up adding a function description and cleaning up the comment a bit. Like you said, not directly related to this function but still applicable to update this comment I think since this function is how we expose the (newly fixed) functionality outside of `Blockstore`